File: Difficulty: Easy/Max Sum Subarray of size K/max-sum-subarray-of-size-k.cpp

/*
 Author: Nazmul Islam
 UsingUsing the Sliding Window Technique 
 O(n) Time and O(1) Space
*/
class Solution {
  public:
    int maxSubarraySum(vector<int>& arr, int k) {
        int currSum;
        currSum = 0;
        //to find window sum
        for(int i = 0; i<k; ++i){
            currSum += arr[i];
        }
        int maxSum = currSum;
        
        //Compute sums
        for(int i = k; i<arr.size(); ++i){
            currSum += arr[i] - arr[i - k];
            maxSum = max(maxSum,currSum);
        }
        return maxSum;
    }
};

File: Difficulty: Easy/Max Sum Subarray of size K/max-sum-subarray-of-size-k.cs
class Solution {
    public int maxSubarraySum(int[] arr, int k) {
        int n = arr.Length;
        int windowSum = 0, res = 0;
        for(int i = 0 ; i < k; ++i)
            windowSum += arr[i];
        
        res = windowSum;
        for(int i = k; i < n; ++i){
            windowSum = windowSum - arr[i-k] + arr[i];
            res = Math.Max(windowSum, res);
        }
        return res;
    }
}